python/tank/bootstrap/manager.py

```diff
@@ -516,7 +516,21 @@ def prepare_engine(self, engine_name, entity):
 
         return path
 
-    def get_pipeline_configurations(self, project):
+    def get_resolved_pipeline_configuration_descriptor(self, project_id=None):
+        """
+        Resolves a pipeline configuration and returns its associated descriptor object.
```

Comment:
Why is a descriptor required? Would just the pipeline configuration is be sufficient? After all, one is interchangeable for the other.

Unless you are expecting people to use the same descriptor with multiple projects?